Products/CMFPlone/browser/iconresolver.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+from OFS.Image import File
+from plone.registry.interfaces import IRegistry
+from Products.CMFCore.interfaces import ISiteRoot
+from zope.component import adapter
+from zope.component import getUtility
+from zope.component.hooks import getSite
+from zope.interface import implementer
+from zope.interface import Interface
+from zope.location.interfaces import LocationError
+from zope.traversing.interfaces import ITraversable
+from zope.publisher.interfaces import IPublishTraverse
+
+
+@implementer(IPublishTraverse)
+class IconBaseResolver:
+
+    prefix = 'plone.staticresources.icon.'
+
+    def __init__(self, context, name):
+        self.context = context
+        self.name = name
+
+    def __call__(self):
+        registry = getUtility(IRegistry)
+        icon = prefix + self.name
+        if icon in registry:
+            return registry[icon]
+        raise LocationError(self.name)
+
+
+class IconURLResolver(IconBaseResolver):
+
+    def __call__(self):
+        value = super().__call__()
+        url = getSite().absolute_url() + '/' + value
+        return url
+
+
+class IconTagResolver(IconBaseResolver):
+
+    def __call__(self):
+        value = super().__call__()
+        if value.endswith('.svg'):
+            file = getSite().restrictedTraverse(value)
+            if isinstance(file, File):
+                raise NotImplementedError(
+                    'Resolve icons stored in database is not yet implemented.'
+                )
+            with open(file.path, 'r') as f:
+                tag = f.read()
+        else:
+            tag = '<img src="{src_tag}" class="{class_tag}" alt="{alt_tag}"/>'
+            tag.format(
+                src_tag = value,
+                class_tag = '',
+                alt_tag = self.name,
+            )            
+        return tag
+
+
+@implementer(ITraversable)
+@adapter(ISiteRoot, Interface)
+class IconBaseTraverser:
+
+    resolver = None
```

Comment:
Do we really need another component here? How do you plan to use this one? Before merging I would like to see some documentation draft to understand the big picture.

Reply:
The big picture is described in the issue #3162. This PR is one of two possible solutions. Please consider PR #3194 as prefered.

Reply:
Thanks!